### ISSUE TITLE
Grafana Shortlink points to localhost instead of domain

### ISSUE THREAD

**Issue by SimonTheLeg** [kubermatic/kubermatic#12332]
### What happened?

When using the Grafana `Shorten URL` feature, `localhost` is displayed instead of the domain

![image](https://github.com/kubermatic/kubermatic/assets/20774623/ba76cb0b-ec51-4f0c-b371-1328d1aad0ae)

### Expected behavior

Link with domain is being generated

### How to reproduce the issue?

1. create new dashboard and save it
2. go to share menue → `Shorten URL`

**Comment by SimonTheLeg** [comment]
From the Grafana code it looks like it is using the `root_url` parameter to generate the short-url links.
https://github.com/grafana/grafana/blob/11ea63d9ddee0bc9348f1a9033b2105a8e54286d/pkg/api/short_url.go#L29
https://github.com/grafana/grafana/blob/11ea63d9ddee0bc9348f1a9033b2105a8e54286d/pkg/setting/setting.go#L541

root_url in regular setups is generated from the domain setting, so setting `grafana.ini.server.domain` should be sufficient. Tested with 

```yaml
# values.yaml
grafana.ini:
  server:
    domain: mytestdomain.com
```

results in
![image](https://github.com/kubermatic/kubermatic/assets/20774623/061d057e-b274-4830-87e6-357f77152f29)


tl;dr should be fine to just set the `grafana.ini.server.domain`

**Comment by SimonTheLeg** [comment]
unfortunately after long 

What we have tried:


**1. Templating `grafana.grafana.ini.server.domain` from the value of `.Values.iap.deployments.grafana.ingress.host`**

Unfortunately it is not possible to use values from other subcharts in a subchart. Concretely you cannot refer to anything from the iap values block inside the grafana chart. See https://helm.sh/docs/chart_template_guide/subcharts_and_globals/#overriding-values-from-a-parent-chart

> As the template engine passes values along, it sets the scope. So for the mysubchart templates, only values specifically for mysubchart will be available in .Values

**2. Using an error func to make grafana.ini.domain a required field**
This is definitely possible using the `required` func from helm. However I don't think this is a particularly good option as we only want to make domain required if grafana is being used with iap or behind another ingress. And when we try to figure out if grafana is installed via an ingress, we run into the issue of point 1) again. Namely we cannot know within the grafana subchart if iap subchart values are being set.
```yaml
# grafana/templates/required-values.yaml
# we need to use index here, because the key is called "grafana.ini"
{{- $_ := required "grafana.ini.domain needs to be set" (index .Values.grafana "grafana.ini" "domain") }}
```

**3. Using [globals](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/#global-chart-values)**
@wurbanski and I decided against making use of globals. We think it would make our values.yaml more brittle as it introduces two places in values.yaml that need to be configured and end-users would need to be made aware of this. This could result in a lot of confusion


--- 

As a result we decided it would be best to document in our mla install guide to set grafana.grafana.ini.server.domain. I will work on a PR for this now
